pynamodb/tests/integration/test_migration.py

```diff
@@ -154,3 +154,105 @@ class Meta:
+    assert True == LBAModel.get('pkey1').flag
+    assert False == LBAModel.get('pkey2').flag
+    assert None == LBAModel.get('pkey3').flag
+
+    # Update a value in the model causing LegacyBooleanAttribute to be deserialized
+    LBAModel.get('pkey1').update_item('value', 'value1', 'PUT')
+    LBAModel.get('pkey2').update_item('value', 'value1', 'PUT')
+    LBAModel.get('pkey3').update_item('value', 'value1', 'PUT')
+
+    # Check we are able to read the flag with LegacyBooleanAttribute
+    assert True == LBAModel.get('pkey1').flag
+    assert False == LBAModel.get('pkey2').flag
+    assert None == LBAModel.get('pkey3').flag
+
+    # Update a value in the model causing LegacyBooleanAttribute to be deserialized
+    LBAModel.get('pkey1').update(actions=[LBAModel.value.set('value2')])
+    LBAModel.get('pkey2').update(actions=[LBAModel.value.set('value2')])
+    LBAModel.get('pkey3').update(actions=[LBAModel.value.set('value2')])
+
+    # Check we are able to read the flag with LegacyBooleanAttribute
+    assert True == LBAModel.get('pkey1').flag
+    assert False == LBAModel.get('pkey2').flag
+    assert None == LBAModel.get('pkey3').flag
+
+
+@pytest.mark.ddblocal
+def test_boolean_attribute_deserialization_compatibility(ddb_url):
+    class BAModel(Model):
+        class Meta:
+            table_name = 'ba_deserialization_test'
+            host = ddb_url
+        id = UnicodeAttribute(hash_key=True)
+        flag = BooleanAttribute(null=True)
+        value = UnicodeAttribute(null=True)
+
+    class LBAModel(Model):
+        class Meta:
+            table_name = 'ba_deserialization_test'
+            host = ddb_url
+        id = UnicodeAttribute(hash_key=True)
+        flag = LegacyBooleanAttribute(null=True)
+        value = UnicodeAttribute(null=True)
+
+    LBAModel.create_table(read_capacity_units=1, write_capacity_units=1)
+
+    # Create an object with a LegacyBooleanAttribute flag
+    LBAModel('pkey1', flag=True, value = 'value0').save()
+    LBAModel('pkey2', flag=False, value = 'value0').save()
+    LBAModel('pkey3', flag=None, value = 'value0').save()
+
+    # Check we are able to read the flag with BooleanAttribute
+    assert True == BAModel.get('pkey1').flag
+    assert False == BAModel.get('pkey2').flag
+    assert None == BAModel.get('pkey3').flag
+
+    # Update a value in the model causing BooleanAttribute to be deserialized
+    BAModel.get('pkey1').update_item('value', 'value1', 'PUT')
+    BAModel.get('pkey2').update_item('value', 'value1', 'PUT')
+    BAModel.get('pkey3').update_item('value', 'value1', 'PUT')
+
+    # Check we are able to read the flag with BooleanAttribute
+    assert True == BAModel.get('pkey1').flag
+    assert False == BAModel.get('pkey2').flag
+    assert None == BAModel.get('pkey3').flag
+
+    # Update a value in the model causing BooleanAttribute to be deserialized
+    BAModel.get('pkey1').update(actions=[BAModel.value.set('value2')])
```

Comment:
would it make sense to perform this set of updates on top of fresh LBAModel saves, instead of after update_item calls? technically update_item could correct the issue for the item, and the update calls would succeed even without the change.

Reply:
that makes sense